profile-widget/qmlprofile.cpp

```diff
@@ -51,7 +52,12 @@ void QMLProfile::paint(QPainter *painter)
 		timer.start();
 
 	// let's look at the intended size of the content and scale our scene accordingly
+	// the ProfileScene code runs with a dpr of 1 and we shrink the target rectangle
+	// to make things fit again - this seems illogical, but appears to work
 	QRect painterRect = painter->viewport();
+	painterRect.setSize(QSize(lrint((double)painterRect.size().width() / devicePixelRatio()),
```

Comment:
Yes, that is ugly.

You could try two things:

1) Remove all these changes and do a `painter->setViewTransformEnabled(false)`. That would mean that we do all the DPR handling.

2) Keep the changes, but remove this conversion and replace `painter->viewport()` by `painter->window()`.  This would mean that Qt does the DPR handling.

At least, that would be my interpretation of the documentation. Could be wrong though.

Is there a way to simulate high-DPI on mobile-on-desktop?

Reply:
Sorta

`QT_DEVICE_PIXEL_RATIO=2.75 ./subsurface-mobile`

That just happens to be the ratio on my PIxel 5. Others show an even `2` or `2.5` or `3.1875`

The initial window size becomes weird, but you can resize and make it work.
Or, better, set it via `SUBSURFACE_MOBILE_WIDTH` and `SUBSURFACE_MOBILE_HEIGHT` to an appropriately scaled value - actually, that seems somewhat broken right now... need to figure out what's wrong there as well. But the `QT_DEVICE_PIXEL_RATIO` works to see the effect

Reply:
Oh - nice. Good for visually impaired people. I used QT_SCALE_FACTOR=2.75.

Adding a `painter->setViewTransformEnabled(false);` before `QRect painterRect = painter->viewport();` seems to solve the issue for me. It is of course difficult to predict how this then looks on physical device.

Reply:
You should get yourself one of those fancy "physical devices"... I hear they aren't very expensive... 🤣

Reply:
Hmm - I just tried that on my device and that get's us right to where we were before my hack.
Did I understand you correctly that I was supposed to revert my last commit and simply add that one line?
That once again gets me just the top left 1/8th of the profile with HUGE fonts...

Reply:
Never mind - ignore my previous comment. I will try to analyze this tomorrow.

Reply:
Yeah, I have a few hours starting now, so I'll try some other approaches that don't take away the whole 'super-sampling' advantage of the high DPI screens... because that's why the results look so crappy with my hack...

Reply:
OK, I think I got it.
The problem is that the painter has a transformation set that scales by the DPR. And that's simply wrong. I'll push my next attempt in a moment, but here's a quick comparison shot taken with mobile on desktop...
<img width="420" alt="Screen Shot 2022-02-27 at 12 57 20 PM" src="https://user-images.githubusercontent.com/237471/155899687-85a0ab75-fe2b-440c-85ca-b549d326c4f3.png">
Left is the new code, right is the hack you so rightfully criticized here...

Reply:
I agree with the fix - that's what I hoped the `setViewTransformEnabled(false)` code would achieve. Obviously I didn't read the docs well enough.

Not sure I agree with the comment - I think that the transform should make the painting code DPR independent, so it is kind of logical (if convoluted). But since we do our own DPR scaling, things explode. That makes me wonder whether it wouldn't be preferred to remove our own DPR scaling.

Reply:
I somehow missed these comments.
I'm merging this as is - but you are right... it almost looks like we should play with removing the scaling ...